ynr/apps/candidates/models/versions.py

```diff
@@ -19,19 +20,26 @@ def get_person_as_version_data(person, new_person=False):
     If new_person is True then skip some DB checks that we know will be
     empty. This should reduce the number of queries needed.
     """
+
+    # Prefetch to reduce the number of queries
+    prefetch_related_objects([person], "tmp_person_identifiers")
+
     from candidates.election_specific import shorten_post_label
 
     result = {}
     result["id"] = str(person.id)
 
+    # Add PersonIdentifier fields in to the global version namespace
+    # TODO: move these to dict to make it easier to merge/split/revert
```

Comment:
A dict will be fine as long as we don't allow dupes on `('person', 'value_type')`, but if a future objective is we want to allow >1 IDs of a given type, would it be better to think of this as a list/array when we get as far as DOing the TODO?

Reply:
Yeah, we can chat about the structure – I'm not sure how it should work, or if having more than one (person, value_type) is even a good idea – what does having >1 `wikipedia` page even mean? Would it be better to have `wikipedia_profile` and `party_wikipedia_page` or whatever and force meaning in to the `value_type`?

If we do allow more than one value_type in the key, the value could always be a list.

I'm partly thinking about making template code easier – a bit silly I know, but doing:

`{{ foo.bar[0] }}`

is so much nicer than doing
```
{% for thing in foo %}
  {% if thing.value_type == "bar" %}
    {{ thing.value[0]
  {% endif %}
{% endfor %}

Reply:
It does depend on the `value_type`. I can see quite a strong case for allowing >1 identifiers of type 'other', for example.

Reply:
Let's punt this out of this PR. I've added check box on the [issue tracking this work](https://github.com/DemocracyClub/yournextrepresentative/issues/198#issuecomment-432239826)